src/util/chains.ts

```diff
@@ -97,7 +131,52 @@ export const ID_TO_PROVIDER = (id: ChainId): string => {
       return process.env.JSON_RPC_PROVIDER_ARBITRUM_ONE!;
     case ChainId.ARBITRUM_RINKEBY:
       return process.env.JSON_RPC_PROVIDER_ARBITRUM_RINKEBY!;
+    case ChainId.POLYGON:
+      return process.env.JSON_RPC_PROVIDER_POLYGON!;
+    case ChainId.POLYGON_MUMBAI:
+      return process.env.JSON_RPC_PROVIDER_POLYGON_MUMBAI!;
     default:
       throw new Error(`Chain id: ${id} not supported`);
   }
 };
+
+function isMatic(
+  chainId: number
+): chainId is ChainId.POLYGON | ChainId.POLYGON_MUMBAI {
+  return chainId === ChainId.POLYGON_MUMBAI || chainId === ChainId.POLYGON;
+}
+
+class MaticNativeCurrency extends NativeCurrency {
+  equals(other: Currency): boolean {
+    return other.isNative && other.chainId === this.chainId;
+  }
+
+  get wrapped(): Token {
+    if (!isMatic(this.chainId)) throw new Error('Not matic');
+    return WRAPPED_NATIVE_CURRENCY[this.chainId]!;
+  }
+
+  public constructor(chainId: number) {
```

Comment:
another readability/consistency nit. I feel like we usually place constructor as the first method?